src/Orleans/Configuration/GlobalConfiguration.cs

```diff
@@ -175,10 +198,57 @@ public bool PrimaryNodeIsRequired
         /// Service Id.
         /// </summary>
         public Guid ServiceId { get; set; }
+
         /// <summary>
         /// Deployment Id.
         /// </summary>
         public string DeploymentId { get; set; }
+
+        #region MultiClusterNetwork
+
+        /// <summary>
+        /// Whether this cluster is configured to be part of a multicluster network
+        /// </summary>
+        public bool HasMultiClusterNetwork
+        {
+            get
+            {
+                return !(string.IsNullOrEmpty(GlobalServiceId) || string.IsNullOrEmpty(ClusterId));
+            }
+        }
+
+        /// <summary>
+        /// Shared Global service Id between different datacenters/deployments.
+        /// </summary>
+        public string GlobalServiceId { get; set; }
```

Comment:
Why would `GlobalServiceId` be different from `ServiceId`. I see that it is used in some placed to indirectly detect whether or not the cluster is running as part of a multi-cluster, but that doesn't strike me as a reason for a different value for GlobalServiceId.